monai/_extensions/gmm/gmm_cuda.cu

```diff
@@ -82,13 +82,19 @@ __global__ void CovarianceReductionKernel(
 
   for (int i = 0; i < MATRIX_COMPONENT_COUNT; i++) {
     float matrix_component = matrix[i];
-
+#ifdef __HIP_PLATFORM_AMD__
+    matrix_component += __shfl_down(matrix_component, 16);
```

Comment:
To reduce the amount of code, and remove near-duplication, perhaps defining macros for unsupported functions would make more sense, eg.

```cpp
#ifdef __HIP_PLATFORM_AMD__
  #define __SHFL_DOWN __shfl_down
  #define __SHFL_XOR __shfl_xor
#else
  #define __SHFL_DOWN __shfl_down_sync
  #define __SHFL_XOR __shfl_xor_sync
#endif
```
Wherever `__shfl_down_sync` or `__shfl_xor_sync` appears replace with this macro, the same can be done for the other unsupported functions. If support for these comes about in the future we can remove the macro.

Reply:
Thanks Eric for the suggestion! I will update and commit soon.